src/ucc/api/ucc.h

```diff
@@ -2009,6 +2009,7 @@ typedef enum ucc_event_type {
 typedef enum ucc_ee_type {
     UCC_EE_CUDA_STREAM = 0,
     UCC_EE_CPU_THREAD,
+    UCC_EE_ROCM_STREAM,
```

Comment:
@edgargabriel @Sergei-Lebedev Is there value in keeping the CUDA stream and ROCM stream separate?

Reply:
@manjugv I think you are right that it might be possible to unify CUDA and ROCM streams, there is just one location (core/ucc_coll.c:367) that uses the stream outside of the corresponding nccl/rccl components. Would you like me to try that?

Reply:
@manjugv @edgargabriel what do you mean by "unifying" CUDA/ROCM streams? Do you mean changing UCC_EE_CUDA_STREAM to UCC_EE_GPU_STREAM in ucc.h? if that is the case, i don't think we can do that since it will break API.

Reply:
I don't think we can unify them because TL UCP supports both CUDA and ROCM and it needs information about underlying execution engine to start correct executor